### PR TITLE
changefeedccl: Augment kafka error message.

### PR DESCRIPTION
**Pull request by miretskiy**
When rejecting a batch of messages due to total batch size, the returned error message was indicating the size of _one_ message, not the batch. This makes it appear that the error message is wrong. With this change, the error message describes the size that is relevant the cause of the rejection.

The error message is typically `kafka server: Message was too large, server rejected it to avoid allocation error`, which can come from either server (Kafka) or client (CRDB using Sarama).

Release Notes: None
Release Justification: low impact, observability improvement.

### CODE REVIEW

**Review by HonoreDB — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/80206)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @miretskiy)

File: pkg/ccl/changefeedccl/sink_kafka.go [hunk 1]
```diff
@@ -362,15 +366,17 @@ func (s *kafkaSink) workerLoop() {
 				// and errors sending dummy messages used to prefetch metadata.
 				if err.Msg != nil && err.Msg.Key != nil && err.Msg.Value != nil {
 					ackError = errors.Wrapf(ackError,
-						"while sending message with key=%s, size=%d",
-						err.Msg.Key, err.Msg.Key.Length()+err.Msg.Value.Length())
+						"while sending message with key=%s, size=%d, stats=%s",
```
Comment (shermanCRL) [inline pkg/ccl/changefeedccl/sink_kafka.go:369]: Label the size units here maybe?

**Comment by HonoreDB** [comment]
blathers backport 22.1

**Pull request merged**